open-api/rest-catalog-open-api.yaml

```diff
@@ -1233,9 +1233,9 @@ components:
         - $ref: '#/components/schemas/BaseUpdate'
         - type: object
           required:
-            - order-id
+            - sort-order-id
           properties:
-            order-id:
+            sort-order-id:
```

Comment:
I can update the corresponding fields for `MetadataUpdate` in another PR.

I'm opening a PR for the last `MetadataUpdate` parsers today.

Reply:
I am moving this change to https://github.com/apache/iceberg/pull/4716, as that's where the associated parser is being written.